lib/src/local_storage.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:sembast/sembast.dart';
+
+import 'constants.dart';
+import 'cross_platform/path.dart' as path;
+import 'cross_platform/sembast.dart';
+
+class LocalStorage {
+  bool _isOpen = false;
+  late final Database _localDb;
+  late final StoreRef _store;
+
+  LocalStorage();
+
+  Future _openDb() async {
+    if (!_isOpen) {
+      _localDb = await getDatabaseFactory().openDatabase(
+          '${path.getBasePath()}/${Constants.persistSessionDbFileName}');
+
+      _store = StoreRef.main();
+
+      _isOpen = true;
+    }
+  }
+
+  Future<String> read(String key) async {
+    await _openDb();
+
+    final value = await _store.record(key).get(_localDb);
+
+    return value as String;
+  }
+
+  Future write(String key, String value) async {
+    await _openDb();
+
+    await _store.record(key).put(_localDb, value);
+  }
+
+  Future delete(String key) async {
+    await _openDb();
+
+    await _store.record(key).delete(_localDb);
+  }
+}
```

Comment:
> Opening a database means loading its content in memory. This is an expensive operation. In an application (flutter, web), the recommended way is to open the database on start and to keep it open during the lifetime of the application.

according to sembast doc, we should not call `_openDb` multiple times also.

Reply:
@phamhieu I've done the reword for hive, because it seems easier to use. However, there is the similar 'openBox' method. It's not being called multiple times though, as there is a check to see if it's already open.

I'll push up the code for you to review, but still need to handle the 'box.close()'